apps/web/src/pages/settings/WebhookPage/WebhookPage.tsx

```diff
@@ -0,0 +1,152 @@
+import { Button, Input, inputStyles } from '@novu/design-system';
+import { FC } from 'react';
+import { Controller } from 'react-hook-form';
+
+import { ClipboardIconButton } from '../../../components';
+import { Timeline } from '../../../components/Timeline';
+import { css, cx } from '../../../styled-system/css';
+import { HStack, Stack } from '../../../styled-system/jsx';
+import { text } from '../../../styled-system/recipes';
+import { SettingsPageContainer } from '../SettingsPageContainer';
+import { Text } from './WebhookPage.shared';
+
+// Unfortunately, a wrapper around TimelineItem prevented any styles from applying, so have to use direct import for now
+import { Timeline as MantineTimeline } from '@mantine/core';
+import { useWebhookPage } from './useWebhookPage';
+import { WebhookClaimStatusDisplay } from './WebhookClaimStatusDisplay';
+import { WebhookAdditionalInformationLink } from './WebhookAdditionalInformationLink';
+
+const codeValueInputClassName = css({
+  '& input': {
+    border: 'none !important',
+    background: 'surface.popover !important',
+    color: 'typography.text.secondary !important',
+    fontFamily: 'mono !important',
+  },
+});
+
+// source: https://www.geeksforgeeks.org/how-to-validate-a-domain-name-using-regular-expression/
+const DOMAIN_REGEX = /^(?!-)[A-Za-z0-9-]+([\-\.]{1}[a-z0-9]+)*\.[A-Za-z]{2,6}$/;
+
+export const WebhookPage: FC = () => {
+  const {
+    claimStatus,
+    handleCheckRecords,
+    isMxRecordRefreshing,
+    envName,
+    handleDomainSubmit,
+    mxRecordClipboard,
+    mailServerDomain,
+    domainControl,
+    isUpdateDnsSettingsLoading,
+    inboundParseDomain,
+  } = useWebhookPage();
+
+  return (
+    <SettingsPageContainer
+      title="Inbound parse webhook"
+      header={
+        <Stack gap="100" className={css({ maxW: '600px' })}>
+          <div>
+            <Text>
+              A webhook that processes all incoming email for a domain or subdomain, parses the contents and
+              attachments, and then POSTs multipart/form-data to a URL that you choose for each environment.
+            </Text>
+          </div>
+          <WebhookClaimStatusDisplay
+            status={claimStatus}
+            handleRefresh={handleCheckRecords}
+            isLoading={isMxRecordRefreshing}
+          />
+        </Stack>
+      }
+    >
+      <Stack gap="200" className={css({ maxW: '960px' })}>
+        <Timeline>
+          <MantineTimeline.Item
+            bullet={1}
+            lineVariant="dashed"
+            title={<label htmlFor="inbound-parse-domain-input">Specify a domain name for {envName} environment</label>}
+          >
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** As mentioned in the comment at the top, I had some issues with doing my own wrapper around the Mantine `Timeline.Item` -- no styles would apply the timeline any more. However, I didn't want to get stuck on this since we'll want to reconsider the Timeline more generically in the future when we extract it to the design system‏